microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/UndeployDeployTest.java

```diff
@@ -0,0 +1,304 @@
+     */
+    @Test
+    @InSequence(60)
+    public void testDeployTwoWarsLogsWarning(@ArquillianResource @OperateOnDeployment(FIRST_DEPLOYMENT) URL firstDeploymentUlr,
+            @ArquillianResource @OperateOnDeployment(SECOND_DEPLOYMENT) URL secondDeploymentUlr) throws Exception {
+        deployer.deploy(FIRST_DEPLOYMENT);
+        deployer.deploy(SECOND_DEPLOYMENT);
+        get(firstDeploymentUlr + "?operation=timeout&context=foobar&fail=true").then()
+                .assertThat()
+                .body(containsString("Fallback Hello, context = foobar"));
+        get(secondDeploymentUlr + "?operation=timeout&context=foobar&fail=true").then()
+                .assertThat()
+                .body(containsString("Fallback Hello, context = foobar"));
+
+        try (final OnlineManagementClient client = ManagementClientProvider.onlineStandalone()) {
+            final LogChecker logChecker = new ModelNodeLogChecker(client, 10, true);
+
+            Assert.assertTrue(logChecker.logMatches(Pattern.compile(".*WFLYMPFTEXT0002.*")));
+        }
+
+        deployer.undeploy(FIRST_DEPLOYMENT);
+        deployer.undeploy(SECOND_DEPLOYMENT);
+    }
+
+    /**
+     * @tpTestDetails Deploy first and then second MP FT application. Both of them are the same (same classes/methods)
+     * @tpPassCrit Verify that MP FT Metrics are the same for both of the deployments
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @InSequence(70)
+    public void testFaultToleranceMetricsAreSummedWithSameDeployments(
+            @ArquillianResource @OperateOnDeployment(FIRST_DEPLOYMENT) URL firstDeploymentUlr,
+            @ArquillianResource @OperateOnDeployment(SECOND_DEPLOYMENT) URL secondDeploymentUlr) throws ConfigurationException {
+        deployer.deploy(FIRST_DEPLOYMENT);
+        deployer.deploy(SECOND_DEPLOYMENT);
+
+        get(firstDeploymentUlr + "?operation=timeout&context=foobar&fail=true").then()
+                .assertThat()
+                .body(containsString("Fallback Hello, context = foobar"));
+        // timeout is still working even though 2nd deployment has disabled it
+        get(secondDeploymentUlr + "?operation=timeout&context=foobar&fail=true").then()
+                .assertThat()
+                .body(containsString("Fallback Hello, context = foobar"));
+
+        ArquillianContainerProperties arqProperties = new ArquillianContainerProperties(
+                ArquillianDescriptorWrapper.getArquillianDescriptor());
+        get("http://" + arqProperties.getDefaultManagementAddress() + ":" + arqProperties.getDefaultManagementPort()
+                + "/metrics").then()
+                        .assertThat()
+                        .body(containsString(
+                                "application_ft_org_jboss_eap_qe_microprofile_fault_tolerance_deployments_v10_HelloService_timeout_invocations_total 2.0"));
+
+        deployer.undeploy(FIRST_DEPLOYMENT);
+        deployer.undeploy(SECOND_DEPLOYMENT);
+    }
+
+    @After
```

Comment:
Final suggestion, can we add the following text as Javadoc here and remove the comment inside? WDYT?
```
The purpose of this @After method is to undeploy all deployments if they are present (there is no error if such deployment does not exist = was not deployed) and clean environment for next test, no matter what happens.
```

Reply:
+1, fixed